test/scripts/products/h24-verify-selfmanaged-apicast-and-custom-policy/h24-verify-selfmanaged-apicast-and-custom-policy.go

```diff
@@ -0,0 +1,576 @@
+		} else {
+			log.Error("Self-managed APIcast API gateway Deployment issue", nil)
+		}
+	}
+
+	log.Info("Self-managed APIcast API gateway - Deployment script Completed")
+
+} //end of main()
+
+// Create customer users in dedicated-admins group
+func createCustomerUsers(customerAdminPassword string) error {
+	log.Info("Creating customer users in dedicated-admins group")
+	command := "PASSWORD=" + customerAdminPassword + " DEDICATED_ADMIN_PASSWORD=" + customerAdminPassword + " ../../../../scripts/setup-sso-idp.sh"
+	err := run_shell_command(command)
+	return err
+}
+
+func createNamespace(ctx context.Context, client k8sclient.Client) error {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: apicastNamespace,
+		},
+		Spec: corev1.NamespaceSpec{},
+	}
+	err := client.Get(ctx, k8sclient.ObjectKey{Name: ns.Name}, ns)
+	if err != nil {
+		log.Info("Creating namespace " + ns.Name)
+		err = client.Create(ctx, ns)
+		if err != nil {
+			log.Error("Error create namespace "+apicastNamespace, err)
+			return err
+		}
+	} else {
+		log.Info("Namespace " + apicastNamespace + " already exists")
+	}
+	err = client.Get(ctx, k8sclient.ObjectKey{Name: ns.Name}, ns)
+	if err != nil {
+		log.Error("Error get namespace"+ns.Name, err)
+		return err
+	}
+	command := "oc project " + apicastNamespace
+	return run_shell_command(command)
+}
+
+// Import APIcast Image
+func importApicastImage() error {
+	log.Info("Import APIcast Image")
+	command := "oc import-image 3scale-amp2/apicast-gateway-rhel8:" + apicastImageStreamTag + " --from=registry.redhat.io/3scale-amp2/apicast-gateway-rhel8:" + apicastImageStreamTag + " --confirm"
+	err := run_shell_command(command)
+	return err
+}
+
+// Get 3scale admin token
+func getThreeScaleAdminToken(ctx context.Context, client k8sclient.Client) (string, error) {
+	log.Info("Get 3scale admin Token")
+	s := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "system-seed",
+		},
+	}
+	err := client.Get(ctx, k8sclient.ObjectKey{Name: s.Name, Namespace: threeScaleNamespace}, s)
+	if err != nil {
+		log.Error("Error get 3Scale admin token ", err)
+		return "", err
+	}
+	accessToken := string(s.Data["ADMIN_ACCESS_TOKEN"])
+	//fmt.Printf(" 🔑  Found 3scale admin token access token: %s\n", accessToken)
+	return accessToken, nil
+}
+
+func getThreeScaleAdminPortal(ctx context.Context, client k8sclient.Client) (string, error) {
+	log.Info("Get 3scale admin Portal")
+	opts := k8sclient.ListOptions{
+		Namespace: threeScaleNamespace,
+	}
+	routes := routev1.RouteList{}
+	err := client.List(ctx, &routes, &opts)
+	if err != nil {
+		log.Info("Error obtaining routes list in " + threeScaleNamespace + " namespace  ")
+		return "", err
+	}
+	for _, route := range routes.Items {
+		if strings.Contains(route.Spec.Host, "3scale-admin") {
+			log.Info("Route: " + route.Name)
+			annotations_map := route.GetObjectMeta().GetAnnotations()
+			admin_portal := annotations_map["zync.3scale.net/host"]
```

Comment:
nit camel case rather than snake case
```suggestion
			annotationsMap := route.GetObjectMeta().GetAnnotations()
			adminPortal := annotations_map["zync.3scale.net/host"]
```

Reply:
Done